Automattic/woocommerce-payments
# [WIP] Transactions reporting API

Fixes #6342

#### Changes proposed in this Pull Request

The purpose of this PR is to include a reporting API for transactions. The main difference between the existing transactions API endpoint and the new one is that this API endpoint is defined by using [WordPress REST API Schema](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/). Besides using the REST API schema, new filters have been added and schema is created for this specific resource which will be handy in the future.





#### Testing instructions

1. Make sure that you have configured WooCommerce Rest API
2. Make sure that you have some transactions in the database if you test locally
3. Execute the following endpoint `wp-json/wc/v3/payments/reports/transactions` and make sure that you have some results
4. Apply any of the filters that are added like order_id_is, date_after,source_device_is, ....

-------------------


**Post merge**
